src/app/components/PortraitVideoModal/index.tsx

```diff
@@ -61,9 +67,54 @@ const PortraitVideoModal = ({ items, onClose }: PortraitVideoModalProps) => {
     };
   }, []);
 
+  const playlistLoadedCallback = (e?: Event) => {
+    const player = window?.embeddedMedia?.api?.players()?.bbcMediaPlayer0;
+
+    if (!player) return;
+
+    // @ts-expect-error - playlist is a custom SMP field
+    const { playlist } = e || {};
+
+    const currentId =
+      playlist?.items?.[0]?.vpid || playlist?.items?.[0]?.versionID;
```

Comment:
Some more destructuring?
```suggestion
    const { playlist } = e || {};
    
    const [currentItem] = playlist?.items

    const currentId =
      currentItem?.vpid || currentItem?.versionID;
```